### PR TITLE
6pm d upload and download feature

### PR DESCRIPTION
**Pull request by jiaweiyu2009**
User Story:
As an admin I can download the CSV file of the course table, so that admins can use this CSV file to maintain and organize the course table.

Acceptance Criteria:
 Only admins can have the ability to CSV download.
 There will be a CSV download button for the admin to download.
 The file downloaded should be in .csv format.
 The CSV file should contain every information from the course table in the website.

### CODE REVIEW

**Comment by jiaweiyu2009** [comment]
Added a button on the course page for admins to download csv file of the course table.

**Comment by pconrad** [comment]
Before the staff can review this PR it needs;
* A filled in description that includes what was changed from a user's perspective ("In this PR, we add a feature that allows the users to ____ so that they can ____")
* At least one review from a member of the team.

**Review by lukelewis66 — CHANGES_REQUESTED** [review]
Trusting that the react-csv package does the job properly. Otherwise everything looks good except for line 34 on Courses.js. 
`key: 'name'`
The label for this (line 35) says Course Number, and I think we have agreed on the naming convention being Course Number. So maybe change the key to 'number' instead of name?

**Comment by jiaweiyu2009** [comment]
In this PR, we add a feature that allows the users to download a CSV file containing all information from the course table, so that they can organize and maintain the courses page.

**Comment by jiaweiyu2009** [comment]
> Before the staff can review this PR it needs;
> 
> * A filled in description that includes what was changed from a user's perspective ("In this PR, we add a feature that allows the users to ____ so that they can ____")
> * At least one review from a member of the team.

Added this description.

**Comment by jiaweiyu2009** [comment]
> Trusting that the react-csv package does the job properly. Otherwise everything looks good except for line 34 on Courses.js.
> `key: 'name'`
> The label for this (line 35) says Course Number, and I think we have agreed on the naming convention being Course Number. So maybe change the key to 'number' instead of name?

Fixed that! Thanks.

**Review by lukelewis66 — DISMISSED** [review]
Looks good. Good job jiawei!

File: src/main/java/edu/ucsb/ucsbcslas/controllers/CourseController.java [hunk 1]
```diff
@@ -28,6 +28,22 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import com.opencsv.CSVWriter;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.http.HttpHeaders;
+import java.io.IOException;
+import java.io.StringWriter;
+
```
Comment (scottpchow23) [inline src/main/java/edu/ucsb/ucsbcslas/controllers/CourseController.java:46]: I don't see where we are using any of these imports in this file; I'd remove these if they are not used.

File: pom.xml [hunk 2]
```diff
@@ -26,7 +26,16 @@
...
+		<dependency>
+			<groupId>com.opencsv</groupId>
+			<artifactId>opencsv</artifactId>
+			<version>5.1</version>
+		</dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <scope>runtime</scope>
+        </dependency>
```
Comment (scottpchow23) [inline pom.xml:38]: We're not using these dependencies, so I'd remove them.

**Review by scottpchow23 — CHANGES_REQUESTED** [review]
Almost there! Just a couple of things to remove.

File: src/main/java/edu/ucsb/ucsbcslas/controllers/CourseController.java (hunk 1 above)
Comment (jiaweiyu2009) [reply src/main/java/edu/ucsb/ucsbcslas/controllers/CourseController.java]: fixed, thanks

File: pom.xml (hunk 2 above)
Comment (jiaweiyu2009) [reply pom.xml]: fixed,thanks

**Comment by pconrad** [comment]
20 pts... It would have been 40 but there's a bug:

Compare:
image

with:

contents of downloaded csv:

image

It's probably an issue of the confusion between the "course name" and "course number" labels.

It might even have worked when you first did it, but other changes in the code base may have broken it in the meantime.

In any case, 10 points if you submit a quick bug fix PR.

**Pull request merged**